lib/sentry/transport.ex

```diff
@@ -49,6 +49,14 @@ defmodule Sentry.Transport do
         Process.sleep(sleep_interval)
         post_envelope_with_retries(client, endpoint, headers, payload, retries_left)
 
+      {:error, _status, _headers, _body} when retries_left != [] ->
```

Comment:
Pattern match causes issues on retry. This was my solution -> open to another idea if you have one @whatyouhide

Reply:
> Pattern match causes issues on retry

What issues? You have control over what `request/4` returns so you can return `{:error, {status, headers, body}}` from there and let this fall into the `{:error, reason}` pattern match?

Reply:
When I pass as a tuple it catches in this new error block for a few tests rather than :request_failure. But after thinking about it. I think returning the full error is probably what we want anyhow. Thx for input!! 


elixir
```
 test post_envelope/2 returns an error if the JSON library crashes when decoding the response (Sentry.TransportTest)
     test/sentry/transport_test.exs:121
     match (=) failed
     code:  assert {:error, {:request_failure, reason}} =
              Transport.post_envelope(envelope, HackneyClient, _retries = [])
     left:  {:error, {:request_failure, reason}}
     right: {
              :error,
              {
                :error,
                %RuntimeError{message: "I'm a really bad JSON library"},
                [{Sentry.TransportTest.CrashingJSONLibrary, :decode, 1, [file: ~c"test/sentry/transport_test.exs", line: 129, error_info: %{module: Exception}]}, {Sentry.Transport, :request, 4, [file: ~c"lib/sentry/transport.ex", line: 67]}, {Sentry.Transport, :post_envelope_with_retries, 5, [file: ~c"lib/sentry/transport.ex", line: 34]}, {Sentry.TransportTest, :"test post_envelope/2 returns an error if the JSON library crashes when decoding the response", 1, [file: ~c"test/sentry/transport_test.exs", line: 139]}, {ExUnit.Runner, :exec_test, 2, [file: ~c"lib/ex_unit/runner.ex", line: 472]}, {:timer, :tc, 2, [file: ~c"timer.erl", line: 270]}, {ExUnit.Runner, :"-spawn_test_monitor/4-fun-1-", 6, [file: ~c"lib/ex_unit/runner.ex", line: 394]}]
              }
            }
```